clouddriver-oracle-bmcs/src/test/groovy/com/netflix/spinnaker/clouddriver/oraclebmcs/provider/view/OracleBMCSSecurityGroupProviderSpec.groovy

```diff
@@ -0,0 +1,184 @@
+
+    when:
+    def results = securityGroupProvider.getAll(false).sort { a, b -> a.name.compareTo(b.name) }
+
+    then:
+    results.size() == 4
+    results.collect { it.inboundRules }.flatten() == []
+  }
+
+  def "get by region"(def region, def res) {
+    setup:
+    def cache = new InMemoryCache()
+    cache.mergeAll(Keys.Namespace.SECURITY_GROUPS.ns, [
+      buildSecGroupCacheData(1, "R1", "A1"),
+      buildSecGroupCacheData(2, "R1", "A1"),
+      buildSecGroupCacheData(3, "R2", "A1"),
+      buildSecGroupCacheData(4, "R2", "A1"),
+    ])
+    def securityGroupProvider = new OracleBMCSSecurityGroupProvider(cache, new ObjectMapper())
+
+    expect:
+    securityGroupProvider.getAllByRegion(true, region).collect { it.name } as Set == res as Set
+
+    where:
+    region | res
+    "R1"   | ["Sec Group 1", "Sec Group 2"]
+    "R2"   | ["Sec Group 3", "Sec Group 4"]
+  }
+
+  def "get by account"(def account, def res) {
+    setup:
+    def cache = new InMemoryCache()
+    cache.mergeAll(Keys.Namespace.SECURITY_GROUPS.ns, [
+      buildSecGroupCacheData(1, "R1", "A1"),
+      buildSecGroupCacheData(2, "R1", "A2"),
+      buildSecGroupCacheData(3, "R2", "A1"),
+      buildSecGroupCacheData(4, "R2", "A2"),
+    ])
+    def securityGroupProvider = new OracleBMCSSecurityGroupProvider(cache, new ObjectMapper())
+
+    expect:
+    securityGroupProvider.getAllByAccount(true, account).collect { it.name } as Set == res as Set
+
+    where:
+    account | res
+    "A1"    | ["Sec Group 1", "Sec Group 3"]
+    "A2"    | ["Sec Group 2", "Sec Group 4"]
+  }
+
+  def "get by account and name"() {
+    setup:
+    def cache = new InMemoryCache()
+    cache.mergeAll(Keys.Namespace.SECURITY_GROUPS.ns, [
+      buildSecGroupCacheData(1, "R1", "A1"),
+      buildSecGroupCacheData(2, "R2", "A2"),
+      buildSecGroupCacheData(1, "R3", "A1"),
+      buildSecGroupCacheData(2, "R4", "A2"),
+    ])
+    def securityGroupProvider = new OracleBMCSSecurityGroupProvider(cache, new ObjectMapper())
+
+    when:
+    def results = securityGroupProvider.getAllByAccountAndName(true, "A1", "Sec Group 1")
+
+    then:
+    results.size() == 2
+    results.find { it.region == "R1" } != null
+    results.find { it.region == "R3" } != null
+  }
+
+  def "get by region and account"(def region, def account, def res) {
+    setup:
+    def cache = new InMemoryCache()
+    cache.mergeAll(Keys.Namespace.SECURITY_GROUPS.ns, [
+      buildSecGroupCacheData(1, "R1", "A1"),
+      buildSecGroupCacheData(2, "R1", "A2"),
+      buildSecGroupCacheData(3, "R1", "A1"),
+      buildSecGroupCacheData(4, "R2", "A1"),
+      buildSecGroupCacheData(5, "R2", "A2"),
+    ])
+    def securityGroupProvider = new OracleBMCSSecurityGroupProvider(cache, new ObjectMapper())
+
+    expect:
+    securityGroupProvider.getAllByAccountAndRegion(true, account, region).collect { it.name } as Set == res as Set
+
+    where:
+    region | account | res
```

Comment:
We haven't been consistent about this in the past, but convention is typically to mark the start of the expected values with `||`. So for this test case, it would be: `region | account || res`. And for the earlier ones it would be: `account || res` and `region || res`.